### ISSUE TITLE
Improvements to documentation / examples

### ISSUE THREAD

**Issue by choldgraf** [SciTools/cartopy#899]
I'm helping somebody transcribe a bunch of R code into python for map plotting, and I have found cartopy to be fairly powerful, but really, really hard to figure out the API. There seem to be a lot of things exposed to users that don't have an example showing them off (e.g., I see that @dopplershift just added a user-friendly way to plot state boundaries).

I think it'd be really useful if there were a collection of examples / "intro to cartopy" tutorial that showed off the many things that this package can do!

I can potentially put some time into this as I figure out how this package works, though it'll be a little bit of time because I'm still figuring things out. Just putting this here for now to see if this is something that folks are already planning to implement.

**Comment by choldgraf** [comment]
Note - some of this could be improved by swapping out the current gallery build with something like sphinx-gallery and displaying titles of examples so they're more informative from that page.

**Comment by choldgraf** [comment]
per #900 a useful example would be creating a geoaxis with `plt.subplots`:

```
import matplotlib.pyplot as plt
import cartopy.crs as ccrs
proj = ccrs.Geostationary()
fig, axes = plt.subplots(2, 2, subplot_kw=dict(projection=proj)))
```

(just leaving this here for future reference)

**Comment by lwasser** [comment]
i'd like to second this request! i'm working on these lessons with @choldgraf and he told me about cartopy. i'd love to see some vignettes and could also potentially contribute once i have a grasp of how it works - i'd potentially like to teach with it. thanks!!

**Comment by lgolston** [comment]
This was addressed by #969
